crates/vhost-user-backend/src/handler.rs

```diff
@@ -441,7 +441,13 @@ where
     }
 
     fn get_protocol_features(&mut self) -> VhostUserResult<VhostUserProtocolFeatures> {
-        Ok(self.backend.protocol_features())
+        let features = self.backend.protocol_features();
+
+        // Enable the `XEN_MMAP` protocol feature for backends if xen feature is enabled.
+        #[cfg(feature = "xen")]
+        let features = features | VhostUserProtocolFeatures::XEN_MMAP;
```

Comment:
If we force the flag here, then we should also check that it has been negotiated correctly in `set_protocol_features()`.

Reply:
> If we force the flag here, then we should also check that it has been negotiated correctly in `set_protocol_features()`.

Yeah, something like [this](https://github.com/rust-vmm/vhost/pull/160/files#diff-45488e39bfb65df325f876039433282983d25c9040a9f8f646535b50eedb6426R437), right ?

Reply:
It is strange to set it here and then check it in the other crate.

Can't we do everything here and check that it is set a few lines down in `VhostUserHandler::set_protocol_features()`?

Or I'm missing somenthig?

Reply:
Ah, makes sense. I have moved them both to the vhost crate as that is more central to everything.